src/Util/ParseUrl.php

```diff
@@ -65,18 +69,20 @@ class ParseUrl
 	 *    array  'images'   => (optional) Array of preview pictures
 	 *    string 'keywords' => (optional) The tags which belong to the content
 	 *
-	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 * @throws HTTPException\InternalServerErrorException
 	 * @see   ParseUrl::getSiteinfo() for more information about scraping
 	 * embeddable content
 	 */
-	public static function getSiteinfoCached($url, $no_guessing = false, $do_oembed = true)
+	public static function getSiteinfoCached($url, $no_guessing = false, $do_oembed = true): array
 	{
-		if ($url == "") {
-			return false;
+		if (empty($url)) {
+			throw new HTTPException\InternalServerErrorException('URL parameter can\'t be empty');
```

Comment:
I'm not a huge fan of throwing an exception when the situation is recoverable. This function is also used for fetching additional information for feed items. So when a feed provided some empty value here, the fetching of the feed would fail - and since the worker job failed it would be repeated over and over again. Also the user wouldn't receive any new feed items from this source.

Reply:
It's about return type expectations. In the current state the method always returns an array, or throw an exception if the provided parameter is invalid. I can make sure the calling code doesn't provide an empty URL.

Reply:
Well, you won, now we return a dummy array when URL is empty.